src/MapViewController.swift

```diff
@@ -13,57 +13,182 @@ import CoreLocation
 import Pelias
 import OnTheRoad
 
+/** 
+  Mapzen Error Enumeration
+  - generalError: The general case for things we're not quite sure what happened.
+  - annotationDoesNotExist: The requested annotation does not exist on the map.
+  - apiKeyNotSet: Your Mapzen API key is not set. Set this using the `MapzenManager` class.
+  - routeDoesNotExist: The requested route does not exist.
+ */
 @objc public enum MZError: Int {
   case generalError, annotationDoesNotExist, apiKeyNotSet, routeDoesNotExist
 }
 
+/// Single Tap Gesture Delegate
 public protocol MapSingleTapGestureDelegate : class {
+  /**
+   Asks the delegate if the map should recognize this single tap and perform default functionality.
+
+   - parameter controller: The MapViewController that wants to recognize the tap.
+   - parameter recognizer: The recognizer that initially recognized the tap.
+   - parameter location: The location on the screen that the tap occured in relative to the bounds of the map.
+   
+   - returns: Return true for default functionality, or false if don't want it recognized (or plan on handling it yourself).
+  */
   func mapController(_ controller: MapViewController, recognizer: UIGestureRecognizer, shouldRecognizeSingleTapGesture location: CGPoint) -> Bool
+
+  /**
+   Informs the delegate the map recognized a single tap gesture.
+
+   - parameter controller: The MapViewController that recognized the tap.
+   - parameter recognizer: The recognizer that recognized the tap.
+   - parameter location: The location on the screen that the tap occured in relative to the bounds of the map.
```

Comment:
Maybe `screen coordinates` would be clearer than `location on the screen` if this returns an x,y pixel value and not a lat,lon?